code/src/AddTask.js

```diff
@@ -0,0 +1,157 @@
+        if (!isEditing) {
+            setInputTask(text)
+        } else {
+           dispatch(tasks.actions.onChangeItemInput(text))
+
+        }
+    }
+
+    const onChangeCategorySelect = (category) => {
+        if (!isEditing) {
+            setCategorySelect(category)
+        } else {
+           dispatch(tasks.actions.onChangeCategory(category))
+        }
+    }
+
+    const onChangeDueDate = (date) => {
+            if (!isEditing) {
+                setDueDate(date)
+            } else  dispatch(tasks.actions.onChangeDate(date))
+                 
+    }
+
+    //Go back to home page
+    const onBackBtnClick = () => {
+        navigate('/');
+    }
+  
+
+    return (
+        <div className="container">
+            <IconButton type="button" onClick={onBackBtnClick}>
+                <img tabIndex='1' src={leftBtn} alt='left arrow icon' />
+             </IconButton>
+            <AddTaskForm onSubmit={(e) => onSubmitNewTask(e)}>
+                <InputTaskContainer>
+                    <InputNewTask 
+                    type='text' 
+                    placeholder="Task title"
+                    value={isEditing ? editInput : inputTask}
+                    onChange={(e) => onChangeItem(e.target.value)}
+                    required              
+                    />
+                    <button type="button">
+                        <img src={folder} alt="folder icon" />
+                    </button>
+                </InputTaskContainer>
+
+                <DateCreated>
+                    <img src={calendar} alt='calendar icon'/> 
+                        <p>
+                            Date created: 
+                            <span> {createDate}</span>
+                        </p>
+                </DateCreated>
+
+                <DueDateInput htmlFor='due-date'>
+                        <img src={alarm} alt='alarm clock icon' />
+                        Due by: 
+                        <input 
+                        type='date' 
+                        id="due-date"
+                        value={isEditing ? editDueDate  : dueDate}
+                        min = {createDate}
+                        max = '01/01/2100'
+                        onChange={(e) => onChangeDueDate(e.target.value)}
+                        />
+                </DueDateInput>
+                <CategorySelector background = {OnSelectColorChange}> 
+                    <select value={isEditing ? editSelect :categorySelect} 
+                            onChange={(e) => onChangeCategorySelect(e.target.value)}>
+                        {categoryOption.map(item =>  SelectOptionDisplay(item))}
+                    </select>
+                </CategorySelector>
+                <NoteTextArea 
+                    placeholder='+ Add note'
+                    rows='6'
+                    >
+                </NoteTextArea>
+                <SubmitTaskBtn type="submit">Submit</SubmitTaskBtn>
+            </AddTaskForm>
+        </div>
```

Comment:
So impressive with all functionality! Maybe you could have divided some of the features in separate components :)